lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -207,8 +207,15 @@ defmodule Mix.Tasks.Deps.Compile do
       {"TERM", "dumb"}
     ]
 
-    cmd = "#{rebar_cmd(dep)} bare compile --paths #{escape_path(lib_path)}"
-    do_command(dep, config, cmd, false, env)
+    rebar = Mix.Rebar.local_rebar_path(:rebar3) || handle_rebar_not_found(dep)
```

Comment:
Dialyzer points out that `Mix.Rebar.local_rebar_path(:rebar3)` always return a binary so it's truthy, so `|| handle_rebar_not_found(dep)` and the whole `handle_rebar_not_found` function are unreachable

Reply:
Thank you, I will push a fix soon. :)